prow/repoowners/endpoint_test.go

```diff
@@ -0,0 +1,218 @@
+			expectedBody: `{"error_message":"Path(s) list is empty or non-existing"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid org",
+			body: &ownersRequest{
+				Org:        "9AA48AEB-30EB-4187-89E7-5FAEACC3EC91",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid repo",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "A1A1693C-082C-41B1-9943-3A89B08A57C0",
+				BaseCommit: "Master",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid basecommit",
+			body: &ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "13202CD5-3068-46D4-8C1D-46F544ED39EF",
+				Paths: []string{
+					"src/dir",
+					"src",
+					"foo",
+					"src/dir/conformance",
+					"docs",
+					"docs/file.md",
+				},
+			},
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Please provide corret git information for Org/Repo and Base commit"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name:         "Empty body",
+			body:         nil,
+			contentType:  "application/json",
+			expectedBody: `{"error_message":"Path(s) list is empty or non-existing"}`,
+			expectedCode: http.StatusBadRequest,
+		},
+	}
+
+	client, cleanup, err := getTestClient(testFiles, true, false, true, nil, nil, nil)
+	if err != nil || client == nil || cleanup == nil {
+		t.Fatalf("Error creating test client: %v.", err)
+	}
+	defer cleanup()
+
+	ownersServer := &OwnersServer{OwnersClient: client}
+
+	for _, test := range tests {
+		t.Logf("Running scenario %q", test.name)
+
+		rawBody, err := json.Marshal(test.body)
+		if err != nil {
+			t.Fatalf("Error Ill-formed testing input %v.", err)
+		}
+
+		req, err := http.NewRequest("POST", "/", bytes.NewBuffer(rawBody))
+		if err != nil {
+			t.Fatalf("Error while running POST query to the EndPoint: %v.", err)
+		}
+		req.Header.Set("Content-Type", test.contentType)
+
+		rr := httptest.NewRecorder()
+
+		ownersServer.ServeHTTP(rr, req)
+		if rr.Code != test.expectedCode {
+			t.Errorf("handler returned wrong status code: got %v want %v",
+				rr.Code, test.expectedCode)
+		}
+
+		if rr.Body.String() != test.expectedBody {
+			t.Errorf("handler returned unexpected body: got %v want %v",
+				rr.Body.String(), test.expectedBody)
+		}
+	}
+}
+func TestCorruptedEndpoint(t *testing.T) {
```

Comment:
How is this endpoint corrected? Need comments in the code to tell the why. Most likely at the point of initialising the endpoint for the test.